partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedPartitioner.java

```diff
@@ -124,7 +124,7 @@ public void configure(Map<String, Object> config) {
           pathFormat,
           "Path format cannot be empty."
       );
-    } else if (delim.equals(pathFormat.substring(pathFormat.length() - delim.length() - 1))) {
+    } else if (!StringUtils.isBlank(delim) && pathFormat.endsWith(delim)) {
```

Comment:
Unless I'm mistaken, this should actually be `endsWith(delim)`, otherwise this looks like an off-by-one error.